contracts/MultiSigWallet.sol

```diff
@@ -276,123 +265,117 @@ contract MultiSigWallet {
         returns(bytes32 transactionId_)
     {
-        //  transactionId_ for same set of parameters in propose and
-        //  confirm flow.
         transactionId_ = keccak256(
             abi.encodePacked(_required, address(this), "changeRequirement")
         );
 
-        proposeTransaction(transactionId_);
-        confirmTransaction(transactionId_);
+        submitTransaction(transactionId_);
 
-        if(isTransactionExecuted(transactionId_)) {
-            // Old requirements i.e. number of required confirmations for
-            // an transaction to be executed is being changed. */
+        if (isTransactionExecuted(transactionId_) == true) {
             required = _required;
             emit RequirementChanged(_required);
         }
-
-        return transactionId_;
     }
 
     /**
-     * @notice Allows to propose or confirm intent for wallet to be revoked
-     *         for a transaction.
+     * @notice Revokes confirmation made before by the caller wallet.
+     *
+     * @dev Function requires:
+     *          - The transaction is not executed.
+     *          - The transaction is confirmed by the caller.
      *
-     * @param _transactionId Transaction ID.
+     * @param _transactionId Transaction id to revoke the previous confirmation.
      *
-     * @return transactionId_ which is unique for each unique request.
+     * @return True in case of success otherwise false.
      */
-    function revokeConfirmation(
+    function revokeConfirmation (
         bytes32 _transactionId
     )
         public
         onlyWallet
-        returns(bool /*success*/)
+        returns(bool)
     {
         require(
-            confirmations[_transactionId].status != 2,
-            "Cannot revoke address(this) transaction because it is already executed"
+            isTransactionExecuted(_transactionId) == false,
+            "Transaction is already executed."
         );
         require(
-            confirmations[_transactionId].isConfirmedBy[msg.sender],
-            "Transaction is not confirmed by address(this) wallet"
+            isConfirmedBy(_transactionId, msg.sender) == true,
+            "Transaction is not confirmed by the caller wallet."
         );
 
-        proposeTransaction(_transactionId);
-        confirmTransaction(_transactionId);
-        if(isTransactionExecuted(_transactionId)){
-            confirmations[_transactionId].isConfirmedBy[msg.sender] = false;
-            emit Revoked(
-                _transactionId,
-                msg.sender
-            );
-        }
+        confirmations[_transactionId].isConfirmedBy[msg.sender] = false;
+        emit Revoked(_transactionId, msg.sender);
 
         return true;
     }
 
-    /**
-     * @dev Returns the confirmation status of a transaction.
-     *
-     * @param _transactionId Transaction ID.
-     *
-     * @return Confirmation status.
-     */
+    /** @notice Returns true if requirement was achieved. */
     function isRequirementAchieved(
```

Comment:
It would give incorrect result for following scenario :-
Consider the requirement is 2 and there are 3-wallets[A, B, C] present.Wallet A proposes an transaction to replace wallet B with wallet D and B also confirms the transaction.As requirement is acheived and transaction is confirmed,wallets present would be A, D, C.Now,when this method is called it will return false for the transaction which was already confirmed and executed. 
Lets discuss solution for this on call.